kratos/sources/model_part.cpp

```diff
@@ -566,6 +574,158 @@ void ModelPart::AddProperties(ModelPart::PropertiesType::Pointer pNewProperties,
     }
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool ModelPart::HasProperties(
+    IndexType PropertiesId,
+    IndexType MeshIndex
+    ) const
+{
+    auto pprop_it = GetMesh(MeshIndex).Properties().find(PropertiesId);
+    if(pprop_it != GetMesh(MeshIndex).Properties().end()) { // Property does exist
+        return true;
+    }
+
+    return false;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+bool ModelPart::RecursivelyHasProperties(
+    IndexType PropertiesId,
+    IndexType MeshIndex
+    ) const
+{
+    auto pprop_it = GetMesh(MeshIndex).Properties().find(PropertiesId);
```

Comment:
why not looking directly in the root modelpart? (I understand that maybe it is not a good idea to change this in this PR)

Reply:
Probably, but this implementation goes level by level instead of going directly into the root model part, probable more robust

Reply:
Do you agree?